File: androidutilslibrary/src/main/java/me/com/androidutilslibrary/UnitUtils/UnitUtils.java
package me.com.androidutilslibrary.UnitUtils;

import android.content.Context;

public class UnitUtils {
    /**
     * @param dps
     * @param context
     * @return
     */
    public static int dpToPx(int dps, Context context) {
        if (context == null) {
            return 0;
        }
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dps * scale + 0.5f);
    }

    /**
     * @param px
     * @param context
     * @return
     */
    public int pxToDp(int px, Context context) {
        if (context == null) {
            return 0;
        }
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) ((px / scale) + 0.5f);
    }
}

File: androidutilslibrary/src/main/java/me/com/androidutilslibrary/unitutils/UnitUtils.kt
package me.com.androidutilslibrary.unitutils

import android.content.Context

object UnitUtils {
    /**
     * @param px      pixel value
     * @param context context
     * @return dp value
     */
    fun pxToDp(px: Int, context: Context): Int {
        val scale = context.resources.displayMetrics.density
        return (px / scale + 0.5f).toInt()
    }

    /**
     * @param dps     dps value
     * @param context context
     * @return return pixel value
     */
    fun dpToPx(dps: Int, context: Context): Int {
        val scale = context.resources.displayMetrics.density
        return (dps * scale + 0.5f).toInt()
    }
}